apps/backend/src/adoption/repositories/adoption-repository.interface.ts

```diff
@@ -1,23 +1,50 @@
-import { AdoptionType, UserType } from '@animavita/models';
-import { AdoptionResponse } from '../adoption.interface';
-import { Location } from '../../user/repositories/mongodb/user.interface';
+import {
+  AnimalAgesType,
+  AnimalGendersType,
+  AnimalSizesType,
+  AnimalType,
+} from '@animavita/types';
+import { UserEntity } from '../../user/repositories/user-repository.interface';
 
-export type IAdoption = AdoptionType & {
-  user: string;
-  location: Location;
+export class AdoptionEntity {
+  id: string;
+  name: string;
+  gender: AnimalGendersType;
+  breed: string;
+  type: AnimalType;
+  age: AnimalAgesType;
```

Comment:
It used to be a number. However, we decided to come up with a generic way of representing the age of the pet since it might be a dog, a cat, or any other animal. Right now I can't think of a better name for that, but it makes sense. Addressing an issue for that https://github.com/animavita/animavita/issues/186